app/services/messages/service.go

```diff
@@ -194,6 +194,24 @@ func (ss *Service) prepareEthereumMintTask(txId string, ethAddress string, amoun
 			ss.logger.Errorf("Failed to establish key transactor. Error %s", err)
 			return
 		}
+
+		if ethTransactor.GasPrice == nil {
+			//Set gas price from memo
+			t, err := ss.transactionRepository.GetByTransactionId(txId)
+			if err != nil {
+				ss.logger.Errorf("Failed to retrive gas price for TX [%s]. Error: %s", txId, err)
+				return
+			}
+			gasPriceGwei, isSuccessful := new(big.Int).SetString(t.GasPriceGwei, 10)
+			if !isSuccessful {
+				ss.logger.Errorf("Failed to parse provided gas price for TX [%s]. Error: %s", txId, err)
+				return
+			}
+			//Convert GWei to Wei
+			mul := new(big.Int).SetUint64(1000000000)
+			ethTransactor.GasPrice = new(big.Int).Mul(gasPriceGwei, mul)
+		}
+
```

Comment:
I do not think it is necessary to get the transaction again. Passing the `gasPrice` as input parameter and then call `ethTransactor.GasPrice = gasPrice` would to the trick.
In `ScheduleEthereumTxForSubmission` `GasPriceGwei` will be retrieved and converted to `wei`.
Let's wait [this](https://github.com/LimeChain/hedera-eth-bridge-validator/issues/139) to be resolved, and then update it.